src/btf.h

```diff
@@ -24,9 +25,35 @@ class BTF
     OK,
   };
 
+  // BTF object for vmlinux or a kernel module.
+  // We're currently storing its name and BTF id.
+  struct BTFObj
+  {
+    struct btf* btf;
+    __u32 id;
+    std::string name;
+
+    BTFObj(struct btf* btf, __u32 id, const std::string& name)
+        : btf(btf), id(id), name(name)
+    {
+    }
```

Comment:
Why do you need this constructor? Designated initializers are better for plain structs:
```
BTFObj foo{.btf = btf, .id = id, .name = name};
```